### ISSUE TITLE
UI - button link element

### ISSUE THREAD

**Issue by xmatthewx** [mozilla/webmaker-android#1549]
depends on #1524

**Comment by vazquez** [comment]
Good progress on the UI for Buttons.

This ticket has several screens so I also set up a quick prototype in InVision: http://invis.io/K22UIEYHU

Here are the screens:

*Scenario:* User taps on Add and chooses Button Element

##### As soon as you add a button, it will display a warning, since the button has an empty URL.

![button add to page](https://cloud.githubusercontent.com/assets/1119821/7396294/0c8ff7c4-ee6e-11e4-871f-8b2533b22844.png)

##### One can either tap on "Tap to Fix" or the Edit button in the screen above to access the Button Options.

![button options](https://cloud.githubusercontent.com/assets/1119821/7396299/10fee0b8-ee6e-11e4-85b7-c739637cbcb3.png)

##### When tapping "Set Link Destination", the user will be asked to pick a page to link to.

![button pick destination page](https://cloud.githubusercontent.com/assets/1119821/7396310/21009966-ee6e-11e4-86d8-d08d586e211a.png)

##### If the user taps on the Bolt they will enter Button Power Link. We give them a choice to a) link to a page in a different project or b) link to an external URL page.

![button power link](https://cloud.githubusercontent.com/assets/1119821/7396313/2348db84-ee6e-11e4-85ae-8e41a18daeb1.png)

##### If they choose to Link to Web, they will see this screen. (I haven't designed the screen after they tap on "Learn More about URL's")

![button link to web](https://cloud.githubusercontent.com/assets/1119821/7396315/25d15c64-ee6e-11e4-83ed-6de05a08880a.png)

##### This is the screen the user sees if they tap on "A page in another project". They will have to choose the Project first, then the page within that project.

![button pick a project](https://cloud.githubusercontent.com/assets/1119821/7396413/d3a53176-ee6e-11e4-90f2-f780a1b92cac.png)

##### Once the user sets a link destination, the button dialogue on the Edit View will look different. It will now invite the user to follow the link.
![button add to page link exists](https://cloud.githubusercontent.com/assets/1119821/7396441/fc1f1c16-ee6e-11e4-8180-6d0263fc6250.png)

Looping @thisandagain and @flukeout

**Comment by vazquez** [comment]
## Updating UI

This ticket has several screens so I also set up a quick prototype in InVision: http://invis.io/K22UIEYHU

Here are the screens:

*Scenario:* User taps on Add and chooses Button Element

##### As soon as you add a button element, a friendly reminder asking you to 'Set Destination' will appear.

![button add to page](https://cloud.githubusercontent.com/assets/1119821/7463344/96555cf8-f26d-11e4-903e-c98d855590b7.png)

##### If the user taps on the Edit button, they will enter the Button Element Options.

![button options](https://cloud.githubusercontent.com/assets/1119821/7463345/9de232c0-f26d-11e4-8e5a-315bc56e1677.png)

##### When tapping "Set Link Destination", the user will be asked to pick a page to link to. Note that tapping on 'Set Destination' also leads to this page.

![button pick destination page](https://cloud.githubusercontent.com/assets/1119821/7463350/a7744b52-f26d-11e4-8c0e-459a1abb5bdc.png)

##### As soon as the user selects a page to link to, we visually display the connection by showing a connection of dotted lines.

![button set destination page](https://cloud.githubusercontent.com/assets/1119821/7463371/c2790302-f26d-11e4-9f5a-eef168d58b62.png)


##### If the user were to tap on the Bolt they will enter Button Power Link. We give them a choice to a) link to a page in a different project or b) link to an external URL page.

![button power link](https://cloud.githubusercontent.com/assets/1119821/7463394/f1d70e8c-f26d-11e4-9f67-0c2c5c0063b5.png)


##### If they choose to Link to Web, they will see this screen. (I haven't designed the screen after they tap on "Learn More about URL's")

![button link to web](https://cloud.githubusercontent.com/assets/1119821/7396315/25d15c64-ee6e-11e4-83ed-6de05a08880a.png)

##### This is the screen the user sees if they tap on "A page in another project". They will have to choose the Project first, then the page within that project.

![button pick a project](https://cloud.githubusercontent.com/assets/1119821/7463402/020fc5fa-f26e-11e4-9145-15bb9a00f484.png)

#### Once the Project is selected the can confirm changes and continue through their selection process.

![button project selected](https://cloud.githubusercontent.com/assets/1119821/7463419/1c558440-f26e-11e4-90fd-4e4a4978275c.png)

##### Once the user sets a link destination, the button dialogue on the Edit View will look different. It will now invite the user to follow the link.

![button add to page link exists](https://cloud.githubusercontent.com/assets/1119821/7463424/2622a084-f26e-11e4-9610-fe4ff956e03e.png)

**Comment by thisandagain** [comment]
Thanks for pulling this together. Right off the bat I feel like this needs to be split into three separate implementations: one that is just within a project (intra-project), one that is between multiple projects (inter-project), and one that is global / tinker mode. Thoughts @xmatthewx ?

**Comment by xmatthewx** [comment]
Yes. Let's definitely land standard mode (intra-project) first in #1645 Beta 1/1. This includes element editor for style & string, and destination picker.  

Tinker mode is listed as Beta 2/2, but not filed yet. URL option seems trivial compared with the complexity of inter-project linking.

**Comment by LauraReynal** [comment]
@LauraReynal  :eyeglasses:

**Comment by xmatthewx** [comment]
This issue is crazy town. After another round of edits, let's post final UI in the separate build tickets.

Unifiy link metaphor:
 - chain link, flag, anchor, rocket, fwd arrow, placemarker, bullseye target,
 - used here: add link, set destination, page origin, destination page, tinker dialog header

Button string:
 - change string. "Ready, go!" ???

Set destination: 
 - large centered origin icon
 - no selected outline around origin
 - remove add-tile placeholders

Tinker:
 - simplify header
 - align header with color tinker view.
 - add bolt

Set URL:
 - change to wikipedia
 - draft copy
  - 'A link can take us to any page or image on the web with an "address" or URL.'
  - 'Links and URLs made the web more powerful than books, radio, and TV. Learn more [MDN].'

Link to other project:
 - try margin on tabs
 - relabel tabs

**Comment by xmatthewx** [comment]
Some things @LauraReynal might test:
 - set link within project
 - using link to nav while in editor
 - edit button text
 - tinker: selecting other projects
 - and of course, _what would you do with links?_

**Comment by xmatthewx** [comment]
Reposting for @vazquez 

### Design Notes

*Scenario:* User taps on Add Content and chooses Button

#### Button Added to Page (No Destination)

- As soon as the user chooses to add a Button, the UI shows a button element selected.
- A small popup appears with `Set Link Destination`.

##### Notes

- When tapping `Set Destination`, the user will immediately go to `Pick Destination Page` (shown below)
- When user taps on the `Edit` button, they will enter `Button Options` (shown below)

![button add to page](https://cloud.githubusercontent.com/assets/1119821/7504843/26721af2-f402-11e4-89e2-be43a474027b.png)

---
 [[unflagged]](http://plzkthx.herokuapp.com)

#### Button Options

- This is the screen that the user arrives after tapping on the `Edit` button.
- This screen allows the user to modify the following:
  1. Set Link Destination
  2. Corners (what we refer to border radius)
  3. Font-family
  4. Background color

##### Notes

- The user is presented with sliders that allow for the modification of values. At the right hand side of the slider, an updating value is shown in px values. The opacity of the value changes if it's not the minimum. As an example, if the Corners value is set at 0px, then we reduce the opacity of the number to decrease the cognitive load of the user. However, as soon as Corners is set to 1px/5px/any value, we change the opacity of the number to 100%.

- To confirm changes, the UI uses the Element Options Navigation in which the user needs to tap the check-mark to confirm changes.

- Border Color has a small circle inside the colour that is presently selected.

- Nice to have: In the case that the options are too long to fit in the screen, it would be really nice if we could hide the content underneath the Preview screen.

![button options](https://cloud.githubusercontent.com/assets/1119821/7505199/5a1a1c8a-f405-11e4-869a-1ed5d026d1a4.png)

---
 [[unflagged]](http://plzkthx.herokuapp.com)

#### Pick Destination Page and Confirm Destination Page

- This is the screen that the user arrives after tapping on `Set Destination`.

##### Notes

1. The current Page is shown with a marker in the center of the page.
2. Tapping on a page will draw a dotted line to its destination.
3. The destination page will show a flag in the center of the page.
4. Once a destination page is shown, a confirmation checkmark appears at the top right hand side.

![button pick destination page](https://cloud.githubusercontent.com/assets/1119821/7505333/4f7b101c-f406-11e4-8654-1e0d35715ebc.png)  

![button set destination page](https://cloud.githubusercontent.com/assets/1119821/7505335/52f8eb24-f406-11e4-9634-e995881ffa6a.png)

---
 [[unflagged]](http://plzkthx.herokuapp.com)

#### Button Tinker Mode

- When in the `Pick Destination Page` screen, if the user taps on Tinker mode :zap:, they enter Button Tinker Mode.

##### Notes

- User is able to:
  1. Link a Page within a project
  2. Link to an external web page
- The modal will disclose if the user taps on the system back button or taps anywhere in the grey area.

![button power link](https://cloud.githubusercontent.com/assets/1119821/7505408/04a5093e-f407-11e4-9f31-ee07b0355844.png)

---
 [[unflagged]](http://plzkthx.herokuapp.com)

#### Link To Web

- The user will arrive at this screen after tapping on `The Web!` in the previous screen.

##### Notes

- The user is able to enter a URL into the textfield.
- When there is a valid entry, the blue checkmark at the top right hand side will appear.

![button link to web](https://cloud.githubusercontent.com/assets/1119821/7505485/d6b7cc90-f407-11e4-8c09-d462075b544a.png)

---
 [[unflagged]](http://plzkthx.herokuapp.com)

#### Button Added to Page (Destination Set)

- Having confirmed all changes, the user can now return to the Page View.

##### Notes

- The text in the small popup modal now reads `Follow Link`. This signifies that a link has already been established.
- If the user taps `Follow Link` they will go to the URL that they previously specified.
- Android browser will take over at this point. Pressing system back button will return the user to this screen.

![button add to page link exists](https://cloud.githubusercontent.com/assets/1119821/7505564/cabe02d2-f408-11e4-9ebb-23ff9146f1a4.png)

**Comment by xmatthewx** [comment]
@vazquez - I think we can call this 3/4 done. 

 - [x] add and edit link
 - [x] set destination page
 - [x] set destination URL
 - [ ] set destination to other project

'Other project' will be implemented in beta 2/2, so this doesn't trump your other priorities. We can bump it back if needed. We need:
 - pick a project view 
  - tap to open project
  - no checkmark, no selected state on project
 - pick a page map view 
  - map with page selected 
  - confirmation checkmark 
  - no zoom or other tools

**Comment by xmatthewx** [comment]
@vazquez - URL settings should reflect tinker mode color and style.

**Comment by vazquez** [comment]
I've updated InVision with the updated screens.

- [x] Pick a project view
- [x] Pick a page map vieew
- [x] URL settings in Tinker mode colour

After updating the URL settings to a Tinker mode header, the pick a Page within a Project seemed out of place with the blue header, since it's also Tinker mode. Therefore:

- [x] Pick a Page in another project in Tinker mode colour
